### ISSUE TITLE
Etcd path for ingress changed in 1.4 

### ISSUE THREAD

**Issue by bprashanth** [kubernetes/kubernetes#32255]
On a 1.3.6 master:
```
# kubectl version
Client Version: version.Info{Major:"1", Minor:"3", GitVersion:"v1.3.6", GitCommit:"ae4550cc9c89a593bcda6678df201db1b208133b", GitTreeState:"clean", BuildDate:"2016-08-26T18:13:23Z", GoVersion:"go1.6.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"3", GitVersion:"v1.3.6", GitCommit:"ae4550cc9c89a593bcda6678df201db1b208133b", GitTreeState:"clean", BuildDate:"2016-08-26T18:06:06Z", GoVersion:"go1.6.2", Compiler:"gc", Platform:"linux/amd64"}

# curl -X GET http://127.0.0.1:4001/v2/keys/registry/ingress/default | python -m json.tool
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1531  100  1531    0     0   172k      0 --:--:-- --:--:-- --:--:--  186k
{
    "action": "get",
    "node": {
        "createdIndex": 246,
        "dir": true,
        "key": "/registry/ingress/default",
        "modifiedIndex": 246,
        "nodes": [
            {
                "createdIndex": 923,
                "key": "/registry/ingress/default/echomap",
                "modifiedIndex": 1059,
                "value": "{\"kind\":\"Ingress\",\"apiVersion\":\"extensions/v1beta1\",\"metadata\":{\"name\":\"echomap\",\"namespace\":\"default\",\"selfLink\":\"/apis/extensions/v1beta1/namespaces/default/ingresses/echomap\",\"uid\":\"1934254f-7576-11e6-8f13-42010af00016\",\"generation\":1,\"creationTimestamp\":\"2016-09-08T03:41:17Z\",\"annotations\":{\"ingress.kubernetes.io/backends\":\"{\\\"k8s-be-30284--b9bff658a8950fa2\\\":\\\"HEALTHY\\\",\\\"k8s-be-30301--b9bff658a8950fa2\\\":\\\"HEALTHY\\\",\\\"k8s-be-30302--b9bff658a8950fa2\\\":\\\"HEALTHY\\\"}\",\"ingress.kubernetes.io/forwarding-rule\":\"k8s-fw-default-echomap--b9bff658a8950fa2\",\"ingress.kubernetes.io/target-proxy\":\"k8s-tp-default-echomap--b9bff658a8950fa2\",\"ingress.kubernetes.io/url-map\":\"k8s-um-default-echomap--b9bff658a8950fa2\"}},\"spec\":{\"backend\":{\"serviceName\":\"echoheadersdefault\",\"servicePort\":80},\"rules\":[{\"host\":\"foo.bar.com\",\"http\":{\"paths\":[{\"path\":\"/foo\",\"backend\":{\"serviceName\":\"echoheadersx\",\"servicePort\":80}}]}},{\"host\":\"bar.baz.com\",\"http\":{\"paths\":[{\"path\":\"/bar\",\"backend\":{\"serviceName\":\"echoheadersy\",\"servicePort\":80}},{\"path\":\"/foo\",\"backend\":{\"serviceName\":\"echoheadersx\",\"servicePort\":80}}]}}]},\"status\":{\"loadBalancer\":{\"ingress\":[{\"ip\":\"130.211.24.48\"}]}}}\n"
            }
        ]
    }
}

# kubectl get ing 
NAME      HOSTS                     ADDRESS         PORTS     AGE
echomap   foo.bar.com,bar.baz.com   130.211.24.48   80        5m
```

But on a 1.4 master: 
```
# kubectl version
Client Version: version.Info{Major:"1", Minor:"3", GitVersion:"v1.3.6", GitCommit:"ae4550cc9c89a593bcda6678df201db1b208133b", GitTreeState:"clean", BuildDate:"2016-08-26T18:13:23Z", GoVersion:"go1.6.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"4+", GitVersion:"v1.4.0-beta.0", GitCommit:"44368db9916cc345ebef8b6fbde3cdf0dc9d79dc", GitTreeState:"clean", BuildDate:"2016-09-01T21:16:48Z", GoVersion:"go1.6.3", Compiler:"gc", Platform:"linux/amd64"}

#  curl -X GET http://127.0.0.1:2379/v2/keys/registry/ingresses/default | python -m json.tool
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1477  100  1477    0     0   150k      0 --:--:-- --:--:-- --:--:--  160k
{
    "action": "get",
    "node": {
        "createdIndex": 6255,
        "dir": true,
        "key": "/registry/ingresses/default",
        "modifiedIndex": 6255,
        "nodes": [
            {
                "createdIndex": 6255,
                "key": "/registry/ingresses/default/echomap",
                "modifiedIndex": 6271,
                "value": "{\"kind\":\"Ingress\",\"apiVersion\":\"extensions/v1beta1\",\"metadata\":{\"name\":\"echomap\",\"namespace\":\"default\",\"selfLink\":\"/apis/extensions/v1beta1/namespaces/default/ingresses/echomap\",\"uid\":\"90271785-7576-11e6-83b9-42010af00002\",\"generation\":1,\"creationTimestamp\":\"2016-09-08T03:44:36Z\",\"annotations\":{\"ingress.kubernetes.io/backends\":\"{\\\"k8s-be-30284--83dd8ff06463057e\\\":\\\"HEALTHY\\\",\\\"k8s-be-30301--83dd8ff06463057e\\\":\\\"HEALTHY\\\"}\",\"ingress.kubernetes.io/forwarding-rule\":\"k8s-fw-default-echomap--83dd8ff06463057e\",\"ingress.kubernetes.io/target-proxy\":\"k8s-tp-default-echomap--83dd8ff06463057e\",\"ingress.kubernetes.io/url-map\":\"k8s-um-default-echomap--83dd8ff06463057e\"}},\"spec\":{\"backend\":{\"serviceName\":\"echoheadersx\",\"servicePort\":80},\"rules\":[{\"host\":\"foo.bar.com\",\"http\":{\"paths\":[{\"path\":\"/foo\",\"backend\":{\"serviceName\":\"echoheadersx\",\"servicePort\":80}}]}},{\"host\":\"bar.baz.com\",\"http\":{\"paths\":[{\"path\":\"/bar\",\"backend\":{\"serviceName\":\"echoheadersy\",\"servicePort\":80}},{\"path\":\"/foo\",\"backend\":{\"serviceName\":\"echoheadersx\",\"servicePort\":80}}]}}]},\"status\":{\"loadBalancer\":{\"ingress\":[{\"ip\":\"130.211.23.32\"}]}}}\n"
            }
        ]
    }
}


# kubectl get ing
NAME      HOSTS                     ADDRESS         PORTS     AGE
echomap   foo.bar.com,bar.baz.com   130.211.23.32   80        2m

```

Obviously this will cause upgrade issues.
@kubernetes/sig-api-machinery ideas? If not I will debug tomorrow.

**Comment by liggitt** [comment]
https://github.com/kubernetes/kubernetes/pull/29557 defaults to using the resource name as the prefix, but ingresses was using a singular form. fix in https://github.com/kubernetes/kubernetes/pull/32258

**Comment by liggitt** [comment]
reopening until cherry-pick merges

**Comment by lavalamp** [comment]
Yikes, good catch. This would have been a disaster.

**Comment by liggitt** [comment]
https://github.com/kubernetes/kubernetes/pull/32294 merged to release-1.4
